command/build.go

```diff
@@ -171,6 +181,29 @@ func (c *BuildCommand) RunContext(buildCtx context.Context, cla *BuildArgs) int
 		c.Ui.Say("Debug mode enabled. Builds will not be parallelized.")
 	}
 
+	// TODO find an option that is not managed by a globally shared Publisher.
+	// This build currently enforces a 1:1 mapping that one publisher can be assigned to a single packer config file.
+	// It also requires that each config type implements this ConfiguredArtifactMetadataPublisher to return a configured bucket.
+	packer.ArtifactMetadataPublisher, diags = packerStarter.ConfiguredArtifactMetadataPublisher()
+	if diags.HasErrors() {
+		return writeDiags(c.Ui, nil, diags)
+	}
+	// TODO we probably want to log if no PAR settings exists at all so adding a TODO to clean this up.
+	if len(diags) > 0 {
+		log.Printf("[TRACE] This doesn't seem to be a Packer Registry enabled build so skipping: %s", diags.Error())
+	}
+
+	if err := packer.ArtifactMetadataPublisher.Initialize(buildCtx); err != nil {
```

Comment:
I think this should be on a per build instantiation phase, in other words, over there:

https://github.com/hashicorp/packer/blob/4be9dfd1838ab09650083549b51a4762ac797fcb/hcl2template/types.packer_config.go#L419-L422

I would wrap each build, and each provisioner and each post-processor into a specific `ArtifactMetadataPublisher`, this way, it would make it easy to know where we're at.

Reply:
Yeah this is do able especially with the rewrite in #11154. Before we can achieve this the GetIterationByFingerprint login has to be implemented. Otherwise in its current state the builds will fail because Initialize only knows how to create new Iterations 😄. I like this suggestion.